Aliases/map - move - over/5eb8d24b-d660-42e5-bbea-47a659bb0827 - Main GVAR.gvar

```diff
@@ -1128,6 +1146,69 @@ if args.last('tokenimport'):
 </drac2>
 
+<drac2>
+if args.last('savebattle') or args.last('deletebattle') or args.last('loadbattle'):
+ userBattles = load_json(get('mapBattles', "{}")) 
+ if args.last('savebattle'):
+  argsBattle = args.last('savebattle')
+  mapBGState = {}
+  for key, value in mapinfo.items():
+   if value:
+    if mapinfo.get('background')!= None:
+     mapBGState["mapbg"] = mapinfo.get('background')
+    mapBGState["mapsize"] = mapsize
+    if mapinfo.get('options')!= None:
+     mapBGState["mapoptions"] = mapinfo.get('options')
+    if mapinfo.get('walls')!= None:
+     mapBGState["walls"] = mapinfo.get('walls').split(', ')
+    if mapinfo.get('objects')!= None:
+     mapBGState["objects"] = mapinfo.get('objects').split('/')
+    break 
+  savedBGState = {"mapattach":mapBGState}
+  savedBGState.update(out)
+  savedBattle = {argsBattle:savedBGState}
+  userBattles.update(savedBattle)
+  set_uvar("mapBattles",dump_json(userBattles))
+  desc.append(f"Saved the battle `{argsBattle.title()}`")
+ if args.last('deletebattle'):
+  argsBattle = args.last('deletebattle')
+  for searchBattle in userBattles:
+   if argsBattle in searchBattle.lower():
+    userBattles.pop(searchBattle)
+    set_uvar("mapBattles",dump_json(userBattles))
+    desc.append(f"Deleted the battle `{searchBattle.title()}`")
+    break
+ if args.last('loadbattle'):
+  argsBattle = args.last('loadbattle')
+  foundBattle = ""
+  for searchBattle in userBattles:
+   if argsBattle in searchBattle.lower():
+    foundBattle = userBattles[searchBattle]
+    for i in foundBattle:
+     if i not in ("mapattach"):
+      foundCombatants = dict(list(foundBattle.items())[1:])
+      if all([combat().get_combatant(i) for i in foundCombatants.keys()]):
+       out = dict(list(foundBattle.items())[1:])
+       foundMapload = foundBattle["mapattach"]
+       if "mapbg" in foundMapload:
+        mapbg = foundMapload["mapbg"]
+       mapsize = foundMapload["mapsize"]
+       if "mapoptions" in foundMapload:
+        mapoptions = foundMapload["mapoptions"]
+       if "walls" in foundMapload:
+        walls = foundMapload["walls"]
+       if "objects" in foundMapload:
+        objects = foundMapload["objects"]
+       setMapview = f"{mapviewlocation}:{mapviewsize}"
+       neweffect = f"""{f"Size: {mapsize}" if mapsize else ""}{f" ~ Background: {mapbg}" if mapbg else ""}{f" ~ Options: {mapoptions}" if mapoptions else ""}{f' ~ Walls: {", ".join(walls)}' if walls else ""}{f' ~ Objects: {"/".join(objects)}' if objects else ""}{f" ~ View: {setMapview}" if setMapview else ""}""".strip(" ~")
+       if mapattach:
+        mapattach.add_effect('map', f"""-attack "||{neweffect}" """) 
+       desc.append(f"Loaded the battle `{argsBattle.title()}`")
+       break  
+      else:
+       desc.append(f"You need to add your combatants to combat first to load their info")
```

Comment:
Does this give the same message if you're missing one combatant as it does if you're missing all of them?

Reply:
Yes. It gives the message until you have all the combatants in initiative that you are trying to load from the same battle.

Reply:
Sounds good, though it might be nice if it listed the missing combatants?